readthedocs/search/api.py

```diff
@@ -160,15 +161,19 @@ def list(self, request, *args, **kwargs):
 
         project_slug = self.request.query_params.get('project', None)
         version_slug = self.request.query_params.get('version', None)
-        query = self.request.query_params.get('q', '')
         total_results = response.data.get('count', 0)
+        time = timezone.now()
+
+        query = self.request.query_params.get('q', '')
+        query = query.lower().strip()
 
         # record the search query with a celery task
         tasks.record_search_query.delay(
             project_slug,
             version_slug,
             query,
             total_results,
+            time,
```

Comment:
Depending on how tasks are serialized, this may not work exactly as planned. I know celery supports Pickle and JSON and I'm not sure which one we are using. Datetimes are not JSON serializable by default although Django does have a way to do it. However, I'm not sure they come back as timezone aware datetimes after they do. Did you verify this?

Reply:
During local development, celery uses `json` serializer.

https://github.com/readthedocs/readthedocs.org/blob/3c04a96b3a9ea8b68ecf270670080285c73c689b/readthedocs/settings/dev.py#L39
(Docs: https://docs.celeryproject.org/en/latest/userguide/configuration.html?highlight=CELERY_RESULT_SERIALIZER#new-lowercase-settings)

I am not sure how to verify this.
I did print the `timezone.now()` and the SQL query -- that looks pretty similar.

timezone.now(): `2019-08-23 08:45:42.841161+00:00`
SQL Query: 
```sql
SELECT "search_searchquery"."id", "search_searchquery"."created", "search_searchquery"."modified", "search_searchquery"."project_id", "search_searchquery"."version_id", "search_searchquery"."query"
FROM "search_searchquery"
INNER JOIN "projects_project"
ON ("search_searchquery"."project_id" = "projects_project"."id")
INNER JOIN "builds_version"
ON ("search_searchquery"."version_id" = "builds_version"."id")
WHERE ("projects_project"."slug" = rtd AND "builds_version"."slug" = latest AND "search_searchquery"."query" LIKE read% ESCAPE '\' AND "search_searchquery"."created" >= 2019-08-23 08:45:32.841161)
ORDER BY "search_searchquery"."created" DESC
```